src/include/homestore/replication/repl_dev.h

```diff
@@ -379,6 +380,21 @@ class ReplDevListener {
     /// @brief Free up user-defined context inside the snapshot_data that is allocated during read_snapshot_data.
     virtual void free_user_snp_ctx(void*& user_snp_ctx) = 0;
 
+    /// @brief handle the leave cluster event
+    // destroy is slightly different from leave_cluster.
+
+    // 1 destory will propose a normal raft log, when this log it committed, it will move the node to destroying state,
+    // and gc thread will stop the raft server and reclaim all the resources allocated to the raft group. there is no
+    // config change, and all the member of the whole raft group will be detructed.
+
+    // 2 well , leave_cluster will propose a config change log, when this log is committed, this node will be removed
+    // from the cluster configuration. so, this callback will only be called on the removed node, which is no longer in
+    // the cluster configuration
+    virtual void on_leave_cluster(const group_id_t& group_id){
```

Comment:
is there a reason we can't bump conan version to 6.6 and make this pure virtual function, and when make HomeObject PR, consume 6.6 homestore to consume this change?

Reply:
the only concern is that before homeobject implements this function, all other PR of homeobject will get a build failure in CI build.